server/routes/allocation.js

```diff
@@ -0,0 +1,85 @@
+const express = require('express');
+const keycloak = require('../keycloak');
+const logger = require('../logger.js');
+const { asyncMiddleware } = require('../error-handler.js');
+const { CreateAllocationSchema } = require('../validation');
+const { expressRequestBodyValidator } = require('../middleware');
+const {
+  createPhaseAllocation,
+  updatePhaseAllocation,
+  getPhaseAllocation,
+} = require('../services/allocations');
+const { FEATURE_PHASE_ALLOCATION } = require('../services/feature-flags');
+
+const router = express.Router();
+
+// Create phase allocation: POST
+router.post(
+  '/',
+  [
+    keycloak.allowRolesMiddleware('ministry_of_health'),
+    keycloak.getUserInfoMiddleware(),
+    expressRequestBodyValidator(CreateAllocationSchema),
+  ],
+  asyncMiddleware(async (req, resp) => {
+    if (!FEATURE_PHASE_ALLOCATION) {
+      return resp.status(501).send('Phase allocation feature not active');
+    }
+    // check if site_phase_allocation exists for site_id and phase_id, if not create one.
+    const { body, hcapUserInfo: user } = req;
+    const allocation = await getPhaseAllocation(body.site_id, body.phase_id);
+    if (allocation) {
+      return resp.status(400).send('An allocation already exists.');
+    }
+    try {
+      const response = await createPhaseAllocation(body, user);
+      logger.info({
+        action: 'allocation_post',
+        performed_by: {
+          username: user.username,
+          id: user.id,
+        },
+        allocationId: response.id,
+      });
+      logger.info(response);
+      return resp.status(201).json(response);
+    } catch (err) {
+      logger.error(err);
+      return resp.status(400).send(`${err}`);
+    }
+  })
+);
+
+// Update phase allocation: PATCH
+router.patch(
+  '/:id',
+  [
+    keycloak.allowRolesMiddleware('ministry_of_health'),
+    keycloak.getUserInfoMiddleware(),
+    expressRequestBodyValidator(CreateAllocationSchema),
```

Comment:
phase_id/site_id are included in the BE validations (not the FE though) 

```
phase_id: yup.number().required('Allocation must be related to a phase'),
site_id: yup.number().required('Allocation must be related to a site'),
```

Reply:
Sorry yes, I was wondering if using the same Schema for the EDIT endpoint works okay since phase_id and site_id is expected, but I don't think we want to / need to be passing those in?

Reply:
ohh good point, I was passing those in but it's not necessary. I'll update the patch/validation for that

Reply:
updated!